cosmwasm/contracts/crosschain-swaps/src/contract.rs

```diff
@@ -51,25 +51,30 @@ pub fn execute(
     info: MessageInfo,
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
+    // IBC transfers support only one token at a time
     match msg {
         ExecuteMsg::OsmosisSwap {
-            input_coin,
+            swap_amount,
             output_denom,
             receiver,
             slippage,
             next_memo,
             on_failed_delivery,
-        } => execute::swap_and_forward(
-            deps,
-            env.block.time,
-            env.contract.address,
-            input_coin,
-            output_denom,
-            slippage,
-            receiver,
-            next_memo,
-            on_failed_delivery,
-        ),
+        } => {
+            let coin = cw_utils::one_coin(&info)?;
```

Comment:
can we set `coin`'s amount to `swap_amount` and pass in just the `coin` to `swap_and_forward`?

Reply:
do you mean pass just the amount?

The reason for passing both is that the amounts may not be the same in the future. Specifically, when we start charging fees for the swap, we will want the amount of funds sent to the contract to be higher than the amount you want to swap

Reply:
https://github.com/osmosis-labs/osmosis/pull/4063/files#r1081482123